### ISSUE TITLE
Best way to deal with junction table and its extra columns

### ISSUE THREAD

**Issue by bpbastos** [samdark/yii2-cookbook#131]
Would be nice a recipe about this subject.

**Comment by samdark** [comment]
What do you want to do with it?

**Comment by samdark** [comment]
No, it's correct place. I've asked because I want to understand your use case to get a better recipe.

**Comment by bpbastos** [comment]
Nice! It's a common problem, unfortunately I haven't found a good way of dealing with this without being too much complex. I've found a good article: http://future500.nl/articles/2013/09/doctrine-2-how-to-handle-join-tables-with-extra-columns/ maybe someone can migrate it to yii2.

**Comment by samdark** [comment]
In general, you need to create a model for that junction table and use it via relations.
